[license] Throw in dev mode after 30 days

oliviertassinari opened the pull request:
A continutation of #9135.

- During the first 30 days, you only get a console.error, as before.
- After 30 days, you get this:

<img width="1004" alt="Screenshot 2023-07-16 at 19 53 37" src="https://github.com/mui/mui-x/assets/3165635/273bae57-3a34-4405-92d7-d65298766afd">

Effectively, if you are doing SSR, you can't build anymore, you would need to remove the license key. I think that the timing for this makes sense now. Developers can opt into the perpetual plan if they prefer this license model.

I worked a bit on #1364 at the same time.

joserodolfofreitas commented:
> Effectively, if you are doing SSR, you can't build anymore, you would need to remove the license key. I think that the timing for this makes sense now. Developers can opt into the perpetual plan if they prefer this license model.

@oliviertassinari, can customers deploy a new production build?

oliviertassinari commented:
> can customers deploy a new production build?

@joserodolfofreitas Yes. I have updated this quote, I wasn't very clear.

I fixed that behavior to be consistent between SSR and client-side in https://github.com/mui/mui-x/pull/10924. Effectively, I missed a bit the objective with the change in this PR.
